src/sentry/monitors/models.py

```diff
@@ -439,6 +439,24 @@ def normalize_before_relocation_import(
         self.guid = uuid4()
         return old_pk
 
+    def ensure_is_muted(self) -> None:
+        """
+        Dual-write: Sync is_muted from monitor environments back to the monitor.
+
+        Sets Monitor.is_muted based on whether all MonitorEnvironments are muted.
+        If there are no environments, is_muted remains unchanged.
+        """
+        # Count total environments and muted environments
+        env_counts = MonitorEnvironment.objects.filter(monitor_id=self.id).aggregate(
+            total=models.Count("id"), muted=models.Count("id", filter=Q(is_muted=True))
+        )
+
+        # Only update if there are environments
+        if env_counts["total"] > 0:
+            all_muted = env_counts["total"] == env_counts["muted"]
+            if self.is_muted != all_muted:
```

Comment:
Probably we could just join these if statements together

Reply:
Yeah. it's fine it's about to change very soon anyway.